src/ring_theory/dedekind_domain.lean

```diff
@@ -0,0 +1,43 @@
+import ring_theory.fractional_ideal
+import ring_theory.ideal.over
+import ring_theory.discrete_valuation_ring
+
+/-- A ring `R` is (at most) one-dimensional if all nonzero prime ideals are maximal. -/
+def ring.is_one_dimensional (R : Type*) [comm_ring R] :=
+∀ p ≠ (⊥ : ideal R), p.is_prime → p.is_maximal
```

Comment:
I should really have called this definition `dimension_le_one` or similar, because fields also satisfy the condition. Since we also have an assumption `is_non_field` in `is_dedekind_domain`, I *think* the best approach is to redefine `is_one_dimensional` to mean that the dimension is exactly one, so we can do away with `is_non_field`. (In either case, I'm going to to write a theorem along the lines of `is_exactly_one_dimensional ↔ is_non_field ∧ (prime -> maximal)`.)